cpp/src/Ice/Service.cpp

```diff
@@ -1455,30 +1433,28 @@ ServiceStatusManager::startUpdate(DWORD state)
 
     _stopped = false;
 
-    _thread = make_shared<StatusThread>(this);
-    _thread->start();
+    _thread = std::thread(&ServiceStatusManager::run, this);
 }
 
 void
 ServiceStatusManager::stopUpdate()
 {
-    IceUtil::ThreadPtr thread;
+    std::thread thread;
 
     {
         lock_guard lock(_mutex);
 
-        if(_thread)
+        if (_thread.get_id() != thread::id())
```

Comment:
Ditto - use joinable?

Reply:
This is to check if we created the thread, joinable would be false if it was created and already finish

Reply:
I was wrong in the comment above, is still joinable until is joined.